autolab/Autolab
# Correctly cache speedgrader symbol tree

## Summary

### Summary generated by Reviewpad on 31 May 23 17:10 UTC
This pull request contains three patches. The first patch corrects the cached HTML object by changing the element ID. The second patch assigns rather than overwrites a new variable for PDF files. The third patch adds back a missing newline at the end of the file.


## Description

- Store contents of `#symbol-tree-container`, rather than only `#symbol-tree-box`, in speedgrader cache
- Assign properties of, rather than redefine, `newFile` for pdf files in `view.js.erb`

## Motivation and Context


The speedgrader's `localCache` does not correctly cache symbol tree contents (loses the enclosing `#symbol-tree-container`), which means that the symbol tree stops updating eventually since `loadFile` operates on `#symbol-tree-container`.

Closes #1697

This PR also fixes a bug introduced in #1840 which causes pdf files to not load in an archive

## How Has This Been Tested?



- Upload an archive file containing files with different symbols and a pdf file
- Switch back and forth between the files
  - Before PR: symbol tree stops updating correctly after a while (e.g. switching away after loading a cached file)
  - After PR: symbol tree continually updates correctly
- Switch to PDF file, ensure it loads

## Types of changes


## Checklist:
